net.sourceforge.pdfchain.json

```diff
@@ -0,0 +1,84 @@
+{
+    "app-id": "net.sourceforge.pdfchain",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "40",
+    "sdk": "org.gnome.Sdk",
+    "command": "pdfchain",
+    "finish-args": [
+    	"--filesystem=xdg-documents",
+        "--filesystem=xdg-download",
+        "--filesystem=xdg-pictures",
+        "--filesystem=home:ro",
+        "--filesystem=host:ro",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--device=dri"
+    ],
+    "cleanup": [
+        "/include",
+        "/lib/pkgconfig",
+        "/share/pkgconfig",
+        "/share/aclocal",
+        "/man",
+        "/share/man",
+        "/share/gtk-doc",
+        "/share/vala",
+        "*.la",
+        "*.a"
+    ],
+    "modules": [
+    	{
+            "name": "pdftk",
+            "buildsystem": "simple",
+            "build-commands": [
+            	"mkdir -p /app/bin",
+                "cp pdftk /app/bin/pdftk",
+                "chmod a+x /app/bin/pdftk"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "__comment1__": "This is a pre-built x86_64 native image of pdftk-java compiled with GraalVM, that's why",
+                    "__comment2__": "flathub build is x86_64 only, see https://gitlab.com/pdftk-java/pdftk#pre-built-binaries",
+                    "url": "https://gitlab.com/pdftk-java/pdftk/-/jobs/924565150/artifacts/raw/build/native-image/pdftk",
+                    "sha256": "6670e053ee6acd9f087ddb5398ff3b77ead352d1a9c7e844e47624844d95ccd5",
+                    "dest-filename": "pdftk"
+                }
+            ]
+        },
+        "gtkmm.json",
+        {
+            "name": "pdfchain",
+            "buildsystem": "simple",
+            "build-commands": [
+            	"sed -i 's,/usr/share/pixmaps/pdfchain.png,/app/share/pixmaps/net.sourceforge.pdfchain.png,g' src/constant.h",
+            	"sed -i 's,Icon=pdfchain,Icon=net.sourceforge.pdfchain,g' application/pdfchain.desktop",
+            	"sed -i 's,Terminal=false,Terminal=false\\nKeywords=pdf;pdfchain;,g' application/pdfchain.desktop",
+            	"./configure --prefix=/app",
+            	"make",
+                "make install"
+            ],
+            "post-install": [
+            	"mv /app/share/pixmaps/pdfchain.png /app/share/pixmaps/net.sourceforge.pdfchain.png",
+                "for i in 16 22 24 32 48 64 96 128 192 256; do
```

Comment:
"rename-icon" doesn't work here?

Reply:
Ok

Reply:
It did not, I tried it, but application didn't show the app icon. "rename-desktop-file" didn't work either, so I renamed that in json too.

Reply:
you probably need a combination of both.

Reply:
Yeah I used both, to no avail.